flexmeasures/data/models/planning/storage.py

```diff
@@ -444,6 +453,16 @@ def deserialize_flex_config(self):
         # Otherwise, we try to retrieve the current state of charge from the asset (if that is the valid one at the start).
         # If that doesn't work, we set the starting soc to 0 (some assets don't use the concept of a state of charge,
         # and without soc targets and limits the starting soc doesn't matter).
+
+        if "soc" in self.flex_model:
+            if (
+                self.flex_model["soc"] is not None
+                and self.flex_model["soc-at-start"] is not None
+            ):
+                raise Exception(
+                    "Both 'soc-at-start' and 'soc' parameters are provided, however, only one of them is necessary."
+                )
```

Comment:
Not yet. I think we can comment this out for now. After this PR, we are using the new `soc` field for saving, but not yet for fetching SoC values. So it should still be possible to use both fields.